lib/packages/docutils/rstgen.nim

```diff
@@ -198,15 +199,35 @@ proc addTexChar(dest: var string, c: char) =
   of '`': add(dest, "\\symbol{96}")
   else: add(dest, c)
 
+proc addJsonChar(dest: var string, c: char) =
+  escapeJsonUnquoted(c, dest)
+
 proc escChar*(target: OutputTarget, dest: var string, c: char) {.inline.} =
   case target
   of outHtml:  addXmlChar(dest, c)
   of outLatex: addTexChar(dest, c)
+  of outJson: addJsonChar(dest, c)
+
+proc makeJsonWFields(kind: string, fields: varargs[tuple[name, data: string]]): string =
+  result &= "{\"kind\":\"" & kind & "\""
```

Comment:
instead, json API's should be reused, operating on `JsonNode`; more readable/editable.
At the very end when a string is needed (eg in addSplitter), use `j.toPretty` or `$j`

Reply:
I know there was a reason why I didn't do this, but I don't quite remember what it was.

Reply:
The reason is that our json module sucks and we need something better.